cmake/Modules/FindSYCL.cmake

```diff
@@ -58,16 +58,36 @@ set(SYCL_HOST_COMPILER "${CMAKE_CXX_COMPILER}"
 if(SYCL_COMPILER)
   set(SYCL_EXECUTABLE ${SYCL_COMPILER} CACHE FILEPATH "SYCL compiler")
 else()
+  if(WIN32)
+    set(SYCL_EXECUTABLE_NAME icx)
+  else()
+    set(SYCL_EXECUTABLE_NAME icpx)
+  endif()
   find_program(SYCL_EXECUTABLE
-    NAMES icpx
+    NAMES ${SYCL_EXECUTABLE_NAME}
     PATHS "${SYCL_PACKAGE_DIR}"
     PATH_SUFFIXES bin bin64
     NO_DEFAULT_PATH
     )
 endif()
 
 set(SYCL_LIBRARIES)
-find_library(SYCL_RUNTIME_LIBRARY sycl HINTS ${SYCL_LIBRARY_DIR})
+# On both Linux and Windows, the current version of SYCL runtime is 7.
+# On Linux, libsycl.so symbolic links to libsycl.so.7.
+# On Windows, sycl7.dll version suffix is used.
+foreach(sycl_lib_version "" 7)
```

Comment:
It violates backward compatibility. When SYCL toolkit uplifts, build will be broken? We should use `libsycl` or `sycl.dll`?

Reply:
Currently, there's no `sycl.dll`. Only `sycl7.dll`, version suffix is used on Windows. `sycl7.dll` on Windows corresponds to `libsycl.so` on Linux. 
In the future, if compiler team adds support to `sycl.dll` on Windows, we can drop this. Otherwise, we need to update `sycl_lib_version` when SYCL runtime version uplifts.

Reply:
Please distinguish WIN32 and Linux. For Linux, the existing implementation is backward compatibility. The change will break it.

Reply:
Thanks, updated.